### PR TITLE
CFM 1414 - Notify user on membership activation with Email

### PR DESCRIPTION
**Pull request by anvmn**
commit 5cc91ecb43b54cf02a818ec6887af4b1934acbe3
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Sun May 29 20:22:21 2016 +0300

     Fixing new property name

commit 55096c6a029b4da0a58cf6d07c665cf66d5487cf
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Sun May 29 20:16:54 2016 +0300

    Integration with new group field - 'Welcome message'

commit d164c1bfc62cf5375448d637abf8e6226a66a255
Merge: 6f30ac6 0f087fb
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Sun May 29 19:46:00 2016 +0300

    Merge branch 'develop' into feature/CFM-604-notify-user

commit 6f30ac63edb85dcff6ce2057f6eb904ae00009ac
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Thu May 26 18:03:54 2016 +0300

    Fixes per code review, take 2.

commit a27af85309c13db8312733036ae2d584d0040590
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Thu May 26 17:19:46 2016 +0300

    Fixes per code review

commit 1119608c2c12d7ac27cf26d07516fdbdf35d9890
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Thu May 26 11:23:22 2016 +0300

    Last bulk of  Coder  fixes

commit 75f292e468842abbcb9ee9d4b3f295cad3b9d23b
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Thu May 26 11:18:39 2016 +0300

    Additional  Coder  fixes

commit 059b3eb35d35c12cc4a2b2ca07f446c0afbc0ce9
Merge: efd8ef7 4c0860a
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 21:11:03 2016 +0300

    Merge branch 'feature/CFM-604-change-members-state' into feature/CFM-604-notify-user

commit efd8ef7ea572217c23c842aaa6d00d83fe49c5c0
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 21:04:51 2016 +0300

    coder_run fixes

commit ab2d7dca8de2d9125d3cdbc24590637737d9e549
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 20:55:05 2016 +0300

    Code review fixes

commit 611881a4691cef2ab100c341af042fa7e86c98a6
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 17:19:05 2016 +0300

    welcome_to_group email template update

commit e042693b273b364bd36755efc99e84f6909ce18a
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 17:10:41 2016 +0300

    Updating 'From' field of welcome_to_group email

commit a2a435a33321a1d3432446f02ab2d4a001dcf173
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 16:43:01 2016 +0300

    Adding field group_welcome_message

commit 7f45a274454e9696fca4cb7fe6c8359e6823782d
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 13:34:48 2016 +0300

    Adding preperations for welcome message token

commit 7ede5a522fec6da310879d09b71a7a9b66a25587
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Wed May 25 13:06:32 2016 +0300

    Token updates
    Message template updates

commit 9fe2c080caa79497d790641340435959a88dc0f3
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Tue May 24 15:20:43 2016 +0300

    Few updates for tokens

commit 40b469d70ff8a6de4d0cca1a7c55164447ed7a33
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Tue May 24 13:47:01 2016 +0300

    Modified message machine name
    Validated mails are sent

commit 378ca7e2c9cd09d945559723e23f589c63db2e5c
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Mon May 23 18:47:04 2016 +0300

    Cosmetical

commit d2ddd1b6bfbcaa8c205b1ef37e0adc89790fe83d
Merge: 858c4f2 4767712
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Mon May 23 18:17:39 2016 +0300

    Merge branch 'feature/CFM-604-change-members-state' into feature/CFM-604-notify-user

commit 858c4f2a4cf95b4377bf3d58a5c3b5b4347d4af8
Author: anatoly vaitsman <anatoly@gizra.com>
Date:   Mon May 23 18:05:37 2016 +0300

    WIP:Mail notification

### CODE REVIEW

**Comment by anvmn** [comment]
https://issuetracker.amplexor.com/jira/browse/CFM-1414

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 1]
```diff
@@ -738,6 +738,16 @@ function c4m_og_open_group_access_activate_pending_members($node) {
...
     $membership->save();
+
+    // Send activation message to member.
+    $message = message_create('c4m_welcome_to_group', array('uid' => $membership->etid));
+    c4m_message_load__c4m_welcome_to_group($message);
+
+    $wrapper = entity_metadata_wrapper('message', $message);
+    $wrapper->field_group_node->set($membership->gid);
+
+    // Message will be saved in message_notify_send_message().
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: Remove this line.

**Comment by ordavidil** [comment]
@anvmn 
Can you please add a screenshot for this one?
thanks.

**Comment by anvmn** [comment]
selection_053

**Pull request merged**